Add wkt support

allevo opened the pull request:
Fix #5 partially.
Tests are missing for the time being: if the implementation seems to be "ok" I'll write them.

allevo commented:
Tests are failing on untouched lines 😢 . It's depends on which rust version are you running maybe.

pjsier commented:
@allevo thanks so much for the PR! The errors are my bad, if it's alright with you I can fix those and update some dependencies today or tomorrow so that the build here will pass.

In the meantime, I was thinking we could use [`wkt::conversion::try_into_geometry`](https://github.com/georust/wkt/blob/2d1370e6aab41df13b29af7dd0e05d875b6ebaee/src/conversion.rs#L60) and use a somewhat simpler setup like we have now in [`handle_shp`](https://github.com/pjsier/echomap/blob/28add9da8566287bbaa23a0789a6da04831ce645/src/main.rs#L161). I should have mentioned that in the issue, so that's my bad, but how does that sound to you?

allevo commented:
Hi!
thanks for helping me!
Yes the macro usage is better than my previous implementation. So I updated the PR accordingly.

I've added a test too.

pjsier commented on src/main.rs:
```diff
@@ -168,6 +171,25 @@ fn handle_shp(file_path: &str, simplification: f64, is_area: bool) -> Vec<GridGe
+    wkt.items
+        .into_iter()
+        .filter_map(|s| try_into_geometry(&s).ok())
+        .flat_map(|geo| {
```
I think `GridGeom::vec_from_geom` should be able to handle the cases here, so we could this instead of the two calls to `flat_map` currently

```rust
.flat_map(|g| GridGeom::<f64>::vec_from_geom(g, simplification, is_area))
```

allevo replied:
yeah! I saw.
Thanks!

pjsier commented on src/main.rs:
```diff
@@ -168,6 +171,25 @@ fn handle_shp(file_path: &str, simplification: f64, is_area: bool) -> Vec<GridGe
         .collect()
 }
 
+fn handle_wkt(input_str: String) -> Vec<GridGeom<f64>> {
```
This should match the signature of `handle_topojson` where we're accepting both `simplification` and `is_area` and then passing them to `GridGeom::vec_from_geom` for processing the geometries

pjsier reviewed (changes requested):
Thanks for the updates! I think we can simplify `handle_wkt` a bit more, but once that's updated it should be good to merge

pjsier commented:
@allevo the changes look great, thanks!
